workflow-manager/main.go

```diff
@@ -254,7 +389,270 @@ func getReadyBatchesGS(ctx context.Context, inputBucket string) ([]string, error
+
+	for _, batchName := range readyBatches {
+		batchPath, err := newBatchPath(batchName)
+		if err != nil {
+			return err
+		}
+		batchIDs = append(batchIDs, batchPath.batchID)
+		batchTimes = append(batchTimes, batchPath.batchDateString())
+
+		// All batches should have the same aggregation ID?
+		if aggregationID != "" && aggregationID != batchPath.aggregationID {
+			return fmt.Errorf("found batch with aggregation ID %s, wanted %s", batchPath.aggregationID, aggregationID)
+		}
+		aggregationID = batchPath.aggregationID
+
+		// Find earliest and latest batch in the aggregation
+		currentBatchTime, err := batchPath.batchDateTime()
+		if err != nil {
+			return err
+		}
+		if earliestBatch == nil {
+			earliestBatch = &batchPath
+		} else {
+			earliestBatchTime, err := earliestBatch.batchDateTime()
+			if err != nil {
+				return err
+			}
+			if currentBatchTime.Before(earliestBatchTime) {
+				earliestBatch = &batchPath
+			}
+		}
+		if latestBatch == nil {
+			latestBatch = &batchPath
+		} else {
+			latestBatchTime, err := latestBatch.batchDateTime()
+			if err != nil {
+				return err
+			}
+			if currentBatchTime.After(latestBatchTime) {
+				latestBatch = &batchPath
+			}
+		}
+	}
+
+	args := []string{
+		"aggregate",
+		"--aggregation-id", aggregationID,
+		"--aggregation-start", earliestBatch.batchDateString(),
+		"--aggregation-end", latestBatch.batchDateString(),
+	}
+	for _, batchID := range batchIDs {
+		args = append(args, "--batch-id")
+		args = append(args, batchID)
+	}
+	for _, batchTime := range batchTimes {
+		args = append(args, "--batch-time")
+		args = append(args, batchTime)
+	}
+
+	jobName := strings.ReplaceAll(fmt.Sprintf("a-%s", earliestBatch.batchDateString()), "/", "-")
+
+	log.Printf("starting aggregation job with args %s", args)
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobName,
+			Namespace: *k8sNS,
+		}, Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					ServiceAccountName: *k8sServiceAccount,
+					RestartPolicy:      "Never",
+					Containers: []corev1.Container{
+						{
+							Args:            args,
+							Name:            "facile-container",
+							Image:           "us.gcr.io/prio-bringup-290620/prio-facilitator:latest",
+							ImagePullPolicy: "Always",
+							EnvFrom: []corev1.EnvFromSource{
+								{
+									ConfigMapRef: &corev1.ConfigMapEnvSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: *aggregateConfigMap,
+										},
+									},
+								},
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name: "BATCH_SIGNING_KEY",
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: *bskSecretName,
+											},
+											Key: "secret_key",
+										},
+									},
+								},
+								{
+									Name: "PACKET_DECRYPTION_KEYS",
```

Comment:
Does the aggregation step use the packet decryption keys? I see that it's an arg on the `aggregate` subcommand, but I don't see it actually used.

Reply:
Yes! We need to decrypt the shares in the original ingestion batch.